File: backend/src/main/java/com/ia/management/model/Student.java
package com.ia.management.model;

import jakarta.persistence.*;

@Entity
@Table(name = "students")
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String regNo;

    @Column(nullable = false)
    private String name;

    private String department;
    private String semester;
    private String section;
    private String phoneNo;
    private String email;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    public Student() {}

    public Student(Long id, String regNo, String name, String department, String semester, String section, String phoneNo, String email, User user) {
        this.id = id;
        this.regNo = regNo;
        this.name = name;
        this.department = department;
        this.semester = semester;
        this.section = section;
        this.phoneNo = phoneNo;
        this.email = email;
        this.user = user;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getRegNo() { return regNo; }
    public void setRegNo(String regNo) { this.regNo = regNo; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getDepartment() { return department; }
    public void setDepartment(String department) { this.department = department; }

    public String getSemester() { return semester; }
    public void setSemester(String semester) { this.semester = semester; }

    public String getSection() { return section; }
    public void setSection(String section) { this.section = section; }

    public String getPhoneNo() { return phoneNo; }
    public void setPhoneNo(String phoneNo) { this.phoneNo = phoneNo; }

    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }

    public User getUser() { return user; }
    public void setUser(User user) { this.user = user; }
}

File: backend-nodejs/src/models/Student.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const User = require('./User');

const Student = sequelize.define('Student', {
    id: {
        type: DataTypes.BIGINT,
        primaryKey: true,
        autoIncrement: true
    },
    regNo: {
        type: DataTypes.STRING,
        unique: true,
        allowNull: false,
        field: 'reg_no'
    },
    name: {
        type: DataTypes.STRING,
        allowNull: false
    },
    department: {
        type: DataTypes.STRING
    },
    semester: {
        type: DataTypes.STRING
    },
    section: {
        type: DataTypes.STRING
    },
    phoneNo: {
        type: DataTypes.STRING,
        field: 'phone_no'
    },
    email: {
        type: DataTypes.STRING
    },
    userId: {
        type: DataTypes.BIGINT,
        field: 'user_id',
        references: {
            model: 'users',
            key: 'id'
        }
    }
}, {
    tableName: 'students',
    timestamps: false
});

// Relationships
Student.belongsTo(User, { foreignKey: 'userId', as: 'user' });
User.hasOne(Student, { foreignKey: 'userId', as: 'student' });

module.exports = Student;
